cmd/cluster/context_test.go

```diff
@@ -0,0 +1,596 @@
+				ID:                tc.clusterID,
+				Name:              tc.clusterName,
+				HypershiftEnabled: tc.hypershiftEnabled,
+				CreationTimestamp: time.Now(),
+			}
+
+			o := &contextOptions{
+				cluster: mockCluster.ToV1Cluster(),
+				infraID: tc.infraID,
+			}
+
+			data := &contextData{
+				OCMEnv: tc.ocmEnv,
+			}
+
+			actualURL := o.buildSplunkURL(data)
+			assert.Equal(t, tc.expectedURL, actualURL, "Generated Splunk URL does not match expected value")
+		})
+	}
+}
+
+func TestPrintOtherLinks(t *testing.T) {
+
+	mockClusterID := "mock-cluster-id"
+	mockExternalClusterID := "mock-external-cluster-id"
+	mockPDServiceID := []string{"PD12345"}
+
+	o := &contextOptions{
+		clusterID:         mockClusterID,
+		externalClusterID: mockExternalClusterID,
+	}
+
+	data := &contextData{
+		pdServiceID: mockPDServiceID,
+	}
+
+	output := captureOutput(func(w io.Writer) {
+		o.printOtherLinks(data, w)
+	})
+
+	expectedLinks := []string{
+		"OHSS Cards",
+		"CCX dashboard",
+		"Splunk Audit Logs",
+		"PagerDuty Service PD12345",
+	}
+
+	for _, link := range expectedLinks {
+		assert.Contains(t, output, link, "Output should contain expected link: %s", link)
+	}
+}
+
+func TestPrintJIRASupportExceptions(t *testing.T) {
+
+	mockIssues := []jira.Issue{
+		{
+			Key: "JIRA-123",
+			Fields: &jira.IssueFields{
+				Type:     jira.IssueType{Name: "Bug"},
+				Priority: &jira.Priority{Name: "High"},
+				Summary:  "Mock issue summary",
+				Status:   &jira.Status{Name: "Open"},
+			},
+		},
+	}
+
+	output := captureOutput(func(w io.Writer) {
+		printJIRASupportExceptions(mockIssues, w)
+	})
+
+	expectedStrings := []string{
+		"- Link: https://issues.redhat.com/browse/JIRA-123",
+	}
+
+	for _, expected := range expectedStrings {
+		assert.Contains(t, output, expected, "Output should contain expected text: %s", expected)
+	}
+}
+
+func TestPrintHistoricalPDAlertSummary(t *testing.T) {
+
+	mockIncidentCounters := map[string][]*pagerduty.IncidentOccurrenceTracker{
+		"PD12345": {
+			{IncidentName: "Network Outage", Count: 3, LastOccurrence: "2024-02-22"},
+			{IncidentName: "Service Downtime", Count: 2, LastOccurrence: "2024-02-20"},
+		},
+	}
+	mockServiceIDs := []string{"PD12345"}
+	mockSinceDays := 7
+
+	output := captureOutput(func(w io.Writer) {
+		printHistoricalPDAlertSummary(mockIncidentCounters, mockServiceIDs, mockSinceDays, w)
+
+	})
+
+	expectedStrings := []string{
+		"PagerDuty Historical Alerts",
+		"Service: https://redhat.pagerduty.com/service-directory/PD12345:",
+		"Type", "Count", "Last Occurrence",
+		"Network Outage", "3", "2024-02-22",
+		"Service Downtime", "2", "2024-02-20",
+		"Total number of incidents [ 5 ] in [ 7 ] days",
+	}
+
+	for _, expected := range expectedStrings {
+		assert.Contains(t, output, expected, "Output should contain expected text: %s", expected)
+	}
+}
+
+func captureOutput(f func(io.Writer)) string {
+	var buf bytes.Buffer
+	f(&buf) // Passing the buffer as an io.Writer
+	return buf.String()
+}
```

Comment:
You don't _really_ need this intermediate step. You can just define your `io.Writer` as a `bytes.Buffer` in tests, pass it to functions, then make assertions against it later. This is just a bit more indirection.

Reply:
Thanks for the suggestion! I simplified the code by removing the intermediate `captureOutput` function and using `bytes.Buffer` directly in the tests as recommended.